### ISSUE TITLE
Location timestamp on Samsung S8 devices is inconsistent

### ISSUE THREAD

**Issue by emc2spider** [mauron85/cordova-plugin-background-geolocation#307]
Not sure this is an actual error of the plugin or just a manufacturer issue, but I'd like to compare with anyone else using this plugin to check if this has happened to someone else already or if it's just me.

When retrieving the location on my Sony Xperia Z5 Compact running Android 6, I always get a consistent timestamp on the location object returned by the plugin whether it is in the foreground or background.

However, when running the same case on a Samsung Galaxy S8+ with Android 7, sometimes the timestamp makes sense i.e. it's a recent timestamp or it gives me values such as 2996969051022 which basically translates to "Sat Dec 20 2064 03:24:11" and this is breaking up the continuity of the location tracking.

## Your Environment

* Plugin version: 2.2.5
* Platform: Android
* OS version: 7.0
* Device manufacturer and model: Samsung Galaxy S8+ (SM-G955F)
* Cordova version (`cordova -v`): 7.0.1
* Cordova platform version (`cordova platform ls`): android 6.2.3
* Plugin configuration options: 
```
            {
                desiredAccuracy: 0,
                stationaryRadius: 1,
                distanceFilter: 1,
                stopOnTerminate: true,
                debug: false,
                locationProvider: backgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER,
                notificationTitle: 'We're tracking your position',
                notificationText: 'The location service is required',
                interval: 5000,
                fastestInterval: 5000,
                stopOnStillActivity: false
            }
```
* Link to your project: N/A

## Context
I'm tracking the location of the user whilst following a predefined route, and animating it on the feed reader end, so I check against the timestamp to determine if I should consider the location as a step forward in the route travel, discarding all older locations since they would potentially make the route traveller jump backwards whenever an old location is sent after a newer one.

## Expected Behavior
All locations should not be older than 5 seconds and they will be progressing on the time as it passes.

## Actual Behavior
Sometimes I get consistent timestamps, others I get timestamps a few decades in the future, this is happening so far in a non deterministic way or at least I haven't figured out a pattern yet. All other Android and iOS devices seem to be working properly. Our app is already live on production and even though the number of users running on this particular device is not majority, this is causing us severe headaches on data reporting and more importantly it's hindering our users experience from both travellers and feed reader sides.

## Possible Fix
Compare the device time against the location time to verify it's validity? If the timestamp is wrong on both the GPS and the device time then we'll just have to add disclaimer notices to our Samsung users and this issue can be closed.

## Steps to Reproduce
1. start the plugin
2. request the location every 5 seconds
3. check the location object timestamp against last reported device date and discard the older
4. Once the date in the future appears everything with the current date will be wrongfully discarded.

## Debug logs
None so far, the app is on production, will get a hold of some later on

**Comment by mauron85** [comment]
Your proposal make lot of sense. From my experience data form gps sensors can have weird values sometimes. Currently the whole chain of shit starts at Samsung, then goes to Android developers, then me and finally it reached you. Meaning nobody decided to filter those apparent wrong gps reads. I've tried, but proper filtering is quite hard to do, because there some many different needs, how people use this plugin.
Sorry for being sarcastic.

**Comment by emc2spider** [comment]
Sarcasm is somewhat appreciated after hunting this elusive situation :)

So far what we're doing around here is clearing the last updated value when we see it's in the future. But not sure how consistent this is. Additionally, we've noticed that on the response from the plugin itself this happens very occasionally. Unlike the Cordova `navigator.geolocation` which throws the error almost exclusively.

Another thing we noticed is that these bugged locations seem to be also fixed in accuracy and the LatLng itself, I'm seriously considering to filter them out completely. However, I'm not sure if this behaviour is deterministic enough to do so.
